planner/core/preprocess.go

```diff
@@ -872,12 +874,14 @@ func (p *preprocessor) checkContainDotColumn(stmt *ast.CreateTableStmt) {
 
 func (p *preprocessor) handleTableName(tn *ast.TableName) {
 	if tn.Schema.L == "" {
-		currentDB := p.ctx.GetSessionVars().CurrentDB
-		if currentDB == "" {
+		if currentDB := p.ctx.GetSessionVars().CurrentDB; currentDB != "" {
+			tn.Schema = model.NewCIStr(currentDB)
+		} else if p.tmpTs != "" {
+			tn.Schema = model.NewCIStr(p.tmpTs)
+		} else {
```

Comment:
It doesn't check whether it's the renamed table name. It will affect all the ALTER TABLE statements. Does they all conform to this rule?

Reply:
I have verified the case of rename, and the case in the original issue. I do not know other cases.

I just checked the source code of mysql, it is of great possibility. The related code is [here](https://github.com/mysql/mysql-server/blob/f8cdce86448a211511e8a039c62580ae16cb96f5/sql/sql_db.cc), specifically the function `mysql_opt_change_db`.

Reply:
@djshow832 Confirmed. It is at this line https://github.com/mysql/mysql-server/blob/7d10c82196c8e45554f27c00681474a9fb86d137/sql/sql_alter.cc#L142. So it conforms to all rules.